### PR TITLE
OBJ-302 Simplified regex to be less vulnerable to reDos attack

### PR DESCRIPTION
**Pull request by ICoxCH**

### CODE REVIEW

File: src/controllers/company.number.controller.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ const formatCompanyNumber = (companyNumber: string, leadPoint: number): string
 // validator middleware that checks for invalid characters in the input
 const postValidators = [
   check(companyNumberFieldName).blacklist(" ").escape().custom((value: string) => {
-    if (!/^[0-9]{8}$|^([a-zA-Z])[0-9]{7}$|^([a-zA-Z]{2})[0-9]{6}$/gm.test(value)) {
+    if (!/^(?=[a-z0-9]{8}$)[a-z]{0,2}\d+$/gmi.test(value)) {
```
Comment (markpit) [inline src/controllers/company.number.controller.ts:50]: just given it a try in https://regex101.com/r/0nFxnL/2
not sure if i have entered it wrong (as it defaults to /gm on end) but uppercase letter didn't match
d1234567 - ok
D1234567 - no match

Comment (markpit) [reply src/controllers/company.number.controller.ts:50]: unless we are changing case in the code?

**Review by mwestacott — APPROVED** [review]

File: src/controllers/company.number.controller.ts (hunk 1 above)
Comment (ICoxCH) [reply src/controllers/company.number.controller.ts:50]: This has gmi on the end, so it should be case insensitive

**Review by markpit — APPROVED** [review]

File: src/controllers/company.number.controller.ts (hunk 1 above)
Comment (mwestacott) [reply src/controllers/company.number.controller.ts:50]: Interesting. Node unit tests use upper-case letters so maybe case is changed in code. Perhaps worth adding tests with lower-case letters too though.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**